packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts

```diff
@@ -241,6 +241,27 @@ export interface AlbControllerOptions {
    * @default - Corresponds to the predefined version.
    */
   readonly policy?: any;
+
+  /**
+   * Override values to be used by the chart.
+   * For nested values use a nested dictionary. For example:
+   * helmChartValues: {
+   *  autoscaling: false,
+   *  ingressClassParams: { create: true }
+   * }
+   *
+   * Note that the following values are set by the controller and cannot be overridden:
+   * - clusterName
+   * - serviceAccount.create
+   * - serviceAccount.name
+   * - region
+   * - vpcId
+   * - image.repository
+   * - image.tag
+   *
+   * @default - No values are provided to the chart.
+   */
+  readonly helmChartValues?: {[key: string]: any};
```

Comment:
I don't think that this is quite the right direction here. From what I can tell, only one or two values need to be overridden for your use case and this is basically giving the user carte blanche to override far more than that. I chatted with @paulhcsun about this PR and I'm thinking that it would likely be better to have a function that can be called after instantiation to disable the load balancer wafv2 behavior instead of adding a prop.

In general, we don't want props to contradict one another or allow values that we know are not allowed (i.e. the list of values you added in the docstring).

What I'm really having trouble with here is understanding how those values are being set in the first place, in your use case. Can you explain what's going on in your specific case a bit more so that our suggestions for the fix don't send you in circles?

Reply:
I would also probably suggest that the function to update this behavior should be scoped to the ALB controller but, again, I'd like to understand better how these fields are being set before saying you should make this code change.

Reply:
hi @TheRealAmazonKendra thanks for your review.

you are correct that for my specific use case I only need to update one or two values, but my intention with this PR is to address the issue of aws-cdk-lib/aws-eks package not exposing the ability to set whatever values the user wants (apart from the ones it is already setting).

the AlbController is just a thin wrapper around the HelmChart construct, yet its constructor's props do not expose any ability to pass values to the underlying helm chart. IMO this was a poor design choice, as I don't think aws-cdk should be the arbiter of which values can be passed to the aws-load-balancer-controller helm chart.

the initial implementation of the Cluster construct did not include any props related to the albController. They were later added as a convenience due to the fact that it is commonly utilized to integrate with aws elbv2. my assumption is that the AlbController construct was defined at the same time, due to the fact that `AlbControllerOptions = Omit<AlbControllerProps, 'cluster'>`. However, the lack of an interface exposing configuration of the helm chart's values means that any user leveraging aws firewall manager either cannot use the "official" construct for the helm chart, or they have to accept that aws-load-balancer-controller will periodically disassociate an fms-applied web acl and their albs willbe unprotected until fms can detect & remediate (usually this is a couple minutes, but we have had occurrences of the remediation failing and albs being unprotected for hours).

Due to the fact that a helm chart's values.yml is the way in which every helm chart's default behavior is overridden, i feel that adding a prop for passing values to the AlbController is a valid design choice. There may be other default behaviors of the helm chart beyond waf management that users may want to disable. Due to the fact that the supported values for aws-load-balancer-controller may vary from one version to the next, I didn't feel it was wise to define a type/interface for a subset of configurable values that will then need to be kept up-to-date. I think if users decide to pass values to the AlbController, they are at the point where they know they need to and we should trust that they are able to decide which values to set.

as far as your suggestion for a function to update values after instantiation, this would need to be exposed on both the AlbController construct and the HelmChart construct. i feel like this is not a great approach and it expands the scope of the solution beyond the AlbController construct.

in short, i feel that this is a fix for an overly-restrictive construct props definition.

Reply:
Hello, I would also want to be able to edit values through this way, in my case for values `enableCertManager`, `cluster.dnsDomain` and some more.
I wonder why this functionality wasn't introduced from the beginning...